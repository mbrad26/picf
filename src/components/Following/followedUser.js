import React, { lazy } from 'react';
// import { Avatar } from '@material-ui/core';
// import { Image } from 'react-bootstrap';

import Avatar from '../Account/avatar';

const Username = lazy(() => import('./username'));

const FollowedUser = ({ user }) => {
  console.log('USER_FOLLOWED: ', user);

  // <Image loading='eager' id='avatar' src={user.avatarUrl} roundedCircle />
  return (
    <>
      <Avatar uid={user.uid} />
      <Username username={user.username} />
    </>
  );
};

export default FollowedUser;
