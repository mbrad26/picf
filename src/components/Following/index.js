import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { Image, Nav } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';

import { doFollowingStatusRequest } from '../../redux/actions/user';

const Following = () => {
  console.log('FOLLOWING');
  const dispatch = useDispatch();
  const { following } = useSelector(state => state.userState);

  useEffect(() => {
    dispatch(doFollowingStatusRequest());
  }, [dispatch]);

  return (
    <>
      <p className='titles'>Following</p>
      {following && 
        following.map(user => 
          <div key={user.uid}>
            <div className='sidebar-avatars'>
              <Nav.Link as={Link} to={`/home/following/${user.uid}`}>
                <Image id='avatar' src={user.avatarUrl} roundedCircle />
                <span> {user.username}</span>
              </Nav.Link>
            </div>
          </div>
        )
      }
    </>
  );
};

export default Following;
