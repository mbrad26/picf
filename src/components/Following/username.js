import React from 'react';

const Username = ({ username }) => 
    <>
      <span> {username}</span>
    </>

export default Username;
