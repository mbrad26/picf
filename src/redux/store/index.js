import { createLogger } from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import { createStore, applyMiddleware } from  'redux';

import rootReducer from '../reducers';
import rootSaga from '../sagas';

const logger = createLogger();
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  {},
  applyMiddleware(saga, logger)
);

saga.run(rootSaga);

export default store;