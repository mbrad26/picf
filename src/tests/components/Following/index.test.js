import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import store from '../../../redux/store';
import Following from  '../../../components/Following/index';

describe('Following', () => {
  it('renders snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Following />
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});