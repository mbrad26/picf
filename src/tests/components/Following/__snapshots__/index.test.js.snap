// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Following renders snapshot 1`] = `
<p
  class="titles"
>
  Following
</p>
`;
